lib/withData.js

```diff
@@ -58,8 +58,10 @@ const getOptions = (options: DataOptions, router: NextRouter) => {
 
   return { variables, query };
 };
-const withData = <T>(
-  ComposedComponent: React.ComponentType<WithDataProps<T>>,
+export const withData = <T>(
+  ComposedComponent: React.ComponentType<WithDataProps<T>> & {
+    getInitialProps?: Object => Promise<Object> | Object
```

Comment:
TODO add typings to this. Didn't care fight more than necessary with flow if some ts is in works.